components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java

```diff
@@ -102,6 +111,11 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
         if(claims == null){
             claims = new HashMap<String,Object>();
         }
+        for (Map.Entry<String, Object> entry : claims.entrySet()) {
+            if(entry.getKey().equals("sub")){
```

Comment:
`"sub".equals(entry.getKey())` could help avoid NPE in the case `entry.getKey()` returns a null value